### ISSUE TITLE
If the error 404 when loading image, Promise returns nothing to the function polyfillImage()

### ISSUE THREAD

**Issue by Drummi42** [chase-moskal/webp-hero#15]
Faced with the problem of stopping processing if you got a picture with a loading error. Maybe I'm wrong in my error, but promis returns nothing if the code is not 200? Thanks in advance for the answer and sorry for the mistakes in words.

```javascript
export async function loadBinaryData(url: string): Promise<Uint8Array> {
	return new Promise<Uint8Array>(function(resolve, reject) {

		const xhr = new XMLHttpRequest()
		xhr.open("GET", url)
		xhr.responseType = "arraybuffer"
	
		xhr.onerror = reject

		xhr.onreadystatechange = () => {
// but if status is 404, we have a infinite hanging promise and  script stopped processing
			if (xhr.readyState == 4 && xhr.status == 200) {
				resolve(new Uint8Array(xhr.response))
			}
		}

		xhr.send()
	})
}
```

**Comment by chase-moskal** [comment]
aye, good catch me matey!

i started working on a solution in pull #16, currently just a rough draft

**Comment by chase-moskal** [comment]
let's keep the issue open until the fix is published

**Comment by chase-moskal** [comment]
now published in `webp-hero@0.0.0-dev.20`

[see changelog](https://github.com/chase-moskal/webp-hero/blob/master/CHANGELOG.md#v000-dev20--2019-04-23)
